packages/tools/pie-wrapper-react/scripts/add-react-wrapper.js

```diff
@@ -102,21 +102,27 @@ export function addReactWrapper (customElementsObject, folderName = process.argv
                 eventsObject = `{\n${eventNames}    }`;
             }
 
+            const componentPropsExportName = `${component.class.name.replace(/^Pie/, '')}Props`;
+
             // Create the main source code
-            componentSrc = `
-import * as React from 'react';
+            componentSrc = `import * as React from 'react';
 import { createComponent${component.class.events?.length > 0 ? ', EventName' : ''} } from '@lit/react';
-import { ${component.class.name} as ${component.class.name}React } from './index';
+import { ${component.class.name} as ${component.class.name}Lit } from './index';
+import { ${componentPropsExportName} } from './defs';
 
 export * from './defs';
 
-export const ${component.class.name} = createComponent({
+const ${component.class.name}React = createComponent({
     displayName: '${component.class.name}',
-    elementClass: ${component.class.name}React,
+    elementClass: ${component.class.name}Lit,
     react: React,
     tagName: '${component.class.tagName}',
     events: ${eventsObject},
 });
+
+type ReactBaseType = Pick<React.ButtonHTMLAttributes<HTMLButtonElement>, 'children' | 'onClick'>
+
+export const ${component.class.name} = ${component.class.name}React as React.ForwardRefExoticComponent<React.PropsWithoutRef<${componentPropsExportName}> & React.RefAttributes<${component.class.name}Lit> & ReactBaseType>;
```

Comment:
I'm curious to know how something like this would behave, rather than setting them all to optional inside defs.ts?
```suggestion
export const ${component.class.name} = ${component.class.name}React as React.ForwardRefExoticComponent<React.PropsWithoutRef<Partial<${componentPropsExportName}>> & React.RefAttributes<${component.class.name}Lit> & ReactBaseType>;
```